### ISSUE TITLE
Request: Turn off printer when temperature is above limit

### ISSUE THREAD

**Issue by loidor** [kantlivelong/OctoPrint-PSUControl#98]
As a safety feature, should a mosfet fail and the hotend or worse, the bed, run with full power and possibly burn stuff up, it would be awesome to have the server cut the power.

**Comment by kantlivelong** [comment]
Hi @loidor,

I like the idea but I'm not entirely decided on if it should be in this plugin. Marking as a request for now.

**Comment by loidor** [comment]
Sure :) I'll probably fork this and hack it in (since I don't think I have a use for waiting until cooled, I'll just swap out some operators), but I thought it might be a useful feature for more people than me.

**Comment by GofX67** [comment]
Thanks for your work, but ...
that would really be a great extension!

Your plugin "knows" the temperatures of hotend and heatbed and in cause of overtemperature (as configured in the menu), simply shut off the printer.

**Comment by ej0rge** [comment]
Bumping this because i think it would be great if there was a way to power off the printer automatically when the printer reports a thermal runaway error.

**Comment by stevenshizzle** [comment]
Temperature Failsafe plugin

**Comment by ej0rge** [comment]
> Temperature Failsafe plugin

That could indeed be used to power off the printer when a temperature is exceeded. 

Personally i was hoping for more flexibility in what octoprint does when the printer reports Error: or !!. Since i have given octoprint the ability to cut the power from the printer.

**Comment by kantlivelong** [comment]
I've decided that this is not a feature I would implement into this plugin especially since the Temperature Failsafe plugin exists. #43 would pair nicely with the Temperatue Failsafe plugin. I'll likely take a look into it again.

I've also looked into providing internal API access to this plugin from other plugins but haven't worked what I want to do with that.

Going to close this specific feature.
